src/connection/host_lookup.c

```diff
@@ -0,0 +1,694 @@
+
+  if (hosts_conf_file_Load == LOAD_INIT)
+    {
+      pthread_mutex_lock (&load_hosts_file_lock);
+      if (hosts_conf_file_Load == LOAD_INIT)
+	{
+	  hosts_conf_file_Load = load_hosts_file ();
+	  if (hosts_conf_file_Load == LOAD_FAIL)
+	    {
+	      pthread_mutex_unlock (&load_hosts_file_lock);
+	      goto return_phase;
+	    }
+	}
+      pthread_mutex_unlock (&load_hosts_file_lock);
+    }
+
+  addr_trans = (struct sockaddr_in *) saddr;
+
+  assert (((hostname != NULL) && (saddr == NULL)) || ((hostname == NULL) && (saddr != NULL)));
+
+  if (lookup_type == IPADDR_TO_HOSTNAME)
+    {
+      if (inet_ntop (AF_INET, &addr_trans->sin_addr, addr_trans_ch_buf, sizeof (addr_trans_ch_buf)) == NULL)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_GENERIC_ERROR, 0);
+	  goto return_phase;
+	}
+
+    }
+
+  /* Look up in the user_host_Map */
+  /* The case which is looking up the IP addr and checking the hostname or IP addr in the hash map */
+  if ((lookup_type == HOSTNAME_TO_IPADDR) && (user_host_Map.find (hostname) != user_host_Map.end ()))
+    {
+      hp = hostent_Cache[user_host_Map.find (hostname)->second];
+    }
+  else if ((lookup_type == IPADDR_TO_HOSTNAME) && (user_host_Map.find (addr_trans_ch_buf) != user_host_Map.end ()))
+    {
+      hp = hostent_Cache[user_host_Map.find (addr_trans_ch_buf)->second];
+    }
+  /*Hostname and IP addr cannot be found */
+  else
+    {
+      if (lookup_type == HOSTNAME_TO_IPADDR)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_UHOST_CANT_LOOKUP_INFO, 1, hostname);
+	}
+      else
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_UHOST_CANT_LOOKUP_INFO, 1, addr_trans_ch_buf);
+	}
+      goto return_phase;
+    }
+
+  return hp;
+
+return_phase:
+
+  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_UHOST_ERROR, 0);
+  fprintf (stdout, "%s\n", er_msg ());
+  fflush (stdout);
+  exit (0);
+}
+
+/*
+ * load_hosts_file () - Load the cubrid_host.conf and be ready for using user_host_Map and hostent_Cache.
+ * 
+ * return   : The result of cubrid_host.conf loading.
+ */
+static int
+load_hosts_file ()
+{
+  FILE *fp;
+  char file_line[LINE_BUF_SIZE + 1];
+  char host_conf_file_full_path[PATH_MAX];
+  char *hosts_conf_dir;
+
+  char *token, temp_token[LINE_BUF_SIZE + 1];
+  char *save_ptr_strtok;
+  /*delimiter */
+  const char *delim = " \t\n";
+  char ipaddr[IPADDR_LEN];
+  char hostname[HOSTNAME_LEN];
+  int cache_idx = 0, temp_idx;
+  /*line muber of cubrid_hosts.conf file */
+  int line_num = 0;
+
+  char addr_trans_ch_buf[IPADDR_LEN];
+  struct in_addr addr_trans;
+
+  /*True, when the string token has hostname, otherwise, string token has IP address */
+  bool hostent_flag;
+
+  memset (file_line, 0, LINE_BUF_SIZE + 1);
+
+  hosts_conf_dir = envvar_confdir_file (host_conf_file_full_path, PATH_MAX, USER_HOSTS_FILE);
+
+  fp = fopen (hosts_conf_dir, "r");
+  if (fp == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_IO_MOUNT_FAIL, 1, host_conf_file_full_path);
```

Comment:
The error message seems to be incorrect.

Reply:
Right. There is not adequate error message.
"fprintf (stderr, "%s[%s]\n", strerror (errno), filename);"
This will be replaced with above example.